File: BOJ/이명환/Silver/[10845]큐.py
import sys
from collections import deque
n = int(sys.stdin.readline())
queue = deque()
for i in range(n):
    instruction  =(sys.stdin.readline().split())
    if instruction[0] == 'push':
        queue.append(instruction[1])
    elif instruction[0] == 'pop':
        if queue:
            print(queue.popleft())
        else:
            print('-1')
    elif instruction[0] == 'size':
        print(len(queue))
    elif instruction[0] == 'empty':
        if queue:
            print('0')
        else:
            print('1')
    elif instruction[0] == 'front':
        if queue:
            print(queue[0])
        else:
            print('-1')
    elif instruction[0] == 'back':
        if queue:
            print(queue[-1])
        else:
            print('-1')

#순수 queue.Queue()로도 풀어볼까 했는데 q[0], q[1] ,q[-1]처럼 인덱싱이 안되네..;

File: BOJ/이명환/Silver/C++/[10845]큐.cpp
// 10845 - 큐

#include <iostream>
#include <queue>
#include <string>

using namespace std;

queue <int> q;

int main(){
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    int N; cin >> N;
    while(N--){
        string command; cin >> command;
        if(command == "push"){
            int X; cin >> X;
            q.push(X);
        }
        else {
            if(command == "pop"){
                if(q.empty()) cout << "-1";
                else {
                    cout << q.front();
                    q.pop();
                }
            }
            else if(command == "size"){
                cout << q.size();
            }
            else if(command == "empty"){
                cout << q.empty();
            }
            else if(command == "front"){
                if(q.empty()) cout << "-1";
                else {
                    cout << q.front();
                }
            }
            else if(command == "back"){
                if(q.empty()) cout << "-1";
                else {
                    cout << q.back();
                }
            }
            cout << '\n';
        }
    }
} 